addons/account/models/sequence_mixin.py

```diff
@@ -47,6 +47,9 @@ def init(self):
                     field=sql.Identifier(self._sequence_field),
                 ))
 
+    def _must_check_constrains_date_sequence(self):
+        return True
```

Comment:
Maybe use that on `account.bank.statement` (see the inheritance of `_constrains_date_sequence`
Or do the same kind of inheritance on `account.move`

Reply:
I'm not sure that it is legit to disable it though, if there is one mistake, the name suggestion will be wrong everytime in the future (might as well not suggest it at all then...)

Reply:
The intention here is that we made manual sequence editing possible because fiduciaries encode bills & invoices not in order all the time. If we don't remove those pesky warnings, the feature is barely usable as-is as they will get warning messages popping all the time despite using the feature as intended.

I think there's still a lot of value for the user to actually propose a value by default, otherwise he will have to craft manually each time which can be a chore. If it gets it wrong once, he can still correct it later right ?

Reply:
> If it gets it wrong once, he can still correct it later right ?

Yes, but it isn't easy to find, and it will poison everything else in the future.

Reply:
Could we get to a middle ground maybe ? No warning if he doesn't follow the sequence number but the format is right, and warning only if he manually enters a number that would change the format ?

Reply:
@chklop So if we have

- BILL/2022/07/0004
- BILL/2022/07/0003
- BILL/2022/07/0002

If the user creates a new bill BILL/2022/07/000**8**, we display no warning.

If the user creates a new bill BLABLA/1, we display the following warning:

> It was previously 'BILL/2022/07/0004' and it is now 'BLABLA/1'.
> The sequence will never restart.
> The incrementing number in this case is '1'.

This is just a warning, so the user will still be able to do it if he/she really wants to.

Yes that can be done, but it is now linked this to this PR comment. This PR comment is about the date constraint which must be deactivated otherwise the user will get the warning each time it encodes its bills in a random order.

Reply:
Right, ok !

Reply:
> It was previously 'BILL/2022/07/0004' and it is now 'BLABLA/1'.

Wait, this is not the constraint related to the code diff.

Reply:
Yep as said here
> Yes that can be done, but it is now linked this to this PR comment. 

~~now~~ not*

Reply:
With regards to the PR comment, I've simply modified `account_bank_statement.py` such that it also overrides `_must_check_constrains_date_sequence` and returns `False` instead of overriding `_constrains_date_sequence` to have consistency between both files

Reply:
Doesn't seem to be done (I liked the solution on bank statements more than this `_must_check_constrains_date_sequence` function btw, I don't like hooks like this when we can easily do without)

Reply:
Bank statement has recently been rewritten and does not use the sequence mixin anymore, so yeah the change has been reverted in the mean time and no longer exists